File: app/src/main/java/com/example/elderapp/adapter/PlaceAdapter.java
package com.example.elderapp.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.example.elderapp.R;

import org.jetbrains.annotations.NotNull;

import java.util.List;





public class PlaceAdapter extends RecyclerView.Adapter<PlaceAdapter.ViewHolder> {

    private List<Place> data;
    private Context context;
    private ItemClickListener mClickListener;

    // data is passed into the constructor
    public PlaceAdapter(Context context, List<Place> data) {
        this.data = data;
        this.context = context;
    }

    // inflates the row layout from xml when needed
    @NotNull
    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_place, parent, false);
        return new ViewHolder(view);
    }

    // binds the data to the TextView in each row
    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        holder.setData(data.get(position));
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    // ViewHolder
    public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

        TextView title, desc;
        ViewHolder(View itemView) {
            super(itemView);
            title = itemView.findViewById(R.id.tv_placeTitle);
            desc =  itemView.findViewById(R.id.tv_placeDesc);
            itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View view) {
            if (mClickListener != null)
                mClickListener.onItemClick(view, getAdapterPosition());
//            Toast.makeText(mContext, title.getText().toString()+"hi", Toast.LENGTH_SHORT).show();
        }
        public void setData(Place data){
            title.setText(data.placeTitle+" -");
            desc.setText(data.placeDesc);
        }
    }


    // convenience method for getting data at click position
    public String getTitle(int id) {
        return data.get(id).placeTitle;
    }
    public String getDesc(int id) {
        return data.get(id).placeDesc;
    }

    // allows clicks events to be caught
    public void setClickListener(ItemClickListener itemClickListener) {
        this.mClickListener = itemClickListener;
    }

    // parent activity will implement this method to respond to click events
    public interface ItemClickListener {
        void onItemClick(View view, int position);
    }
}

File: app/src/main/java/com/example/elderapp/adapter/PlaceAdapter.kt
package com.example.elderapp.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.elderapp.R

class PlaceAdapter(private val data: MutableList<Place>) : RecyclerView.Adapter<PlaceAdapter.ViewHolder?>() {
    private var mClickListener: ItemClickListener? = null

    // inflates the row layout from xml when needed
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_place, parent, false)
        return ViewHolder(view)
    }

    // binds the data to the TextView in each row
    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.setData(data[position])
    }

    override fun getItemCount(): Int {
        return data.size
    }

    // ViewHolder
    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView){
        private var title = itemView.findViewById<TextView>(R.id.tv_name)
        private var desc = itemView.findViewById<TextView>(R.id.tv_phone)
        private var delete = itemView.findViewById<ImageView>(R.id.img_delete)
        private var icon = itemView.findViewById<ImageView>(R.id.img_headshot)


        fun setData(data: Place) {
            title.text = data.title
            desc.text = data.description
            when(data.iconID){
                1 -> icon.setImageResource(R.drawable.ic_place_home)
                2 -> icon.setImageResource(R.drawable.ic_place_park)
                3 -> icon.setImageResource(R.drawable.ic_place_people)
                4 -> icon.setImageResource(R.drawable.ic_place_hospital)
                5 -> icon.setImageResource(R.drawable.ic_place_coffee)
            }

            delete.setOnClickListener {
                if (mClickListener != null) mClickListener!!.onItemClick(adapterPosition)
            }
        }


    }

    // convenience method for getting data at click position
    fun getTitle(id: Int): String {
        return data[id].title
    }

    fun getId(id: Int): Int {
        return data[id].id
    }

    // allows clicks events to be caught
    fun setClickListener(itemClickListener: ItemClickListener?) {
        mClickListener = itemClickListener
    }

    // parent activity will implement this method to respond to click events
    interface ItemClickListener {
        fun onItemClick(position: Int)
    }
}